tests/Microsoft.Bot.Builder.Core.Extensions.Tests/ShowTyping_MiddlewareTests.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder.Adapters;
+using Microsoft.Bot.Schema;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Bot.Builder.Core.Extensions.Tests
+{
+    [TestClass]
+    public class ShowTyping_MiddlewareTests
+    {
+        [TestMethod]
+        [TestCategory("Middleware")]
+        public async Task ShowTyping_TestMiddleware_10SecondResponse_2SecondInterval()
+        {
+            TestAdapter adapter = new TestAdapter()
+                .Use(new ShowTypingMiddleware(500, 2000));
+            
+            await new TestFlow(adapter, async (context) =>
+                {
+                    Thread.Sleep(10000);
+                    await context.SendActivity(context.Request.CreateReply("Message sent after delay"));
+                    await Task.CompletedTask;
+                })
+                .Send("foo")
+                .AssertReply(ValidateTypingActivity, "check typing activity")
+                .AssertReply(ValidateTypingActivity, "check typing activity")
+                .AssertReply(ValidateTypingActivity, "check typing activity")
+                .AssertReply(ValidateTypingActivity, "check typing activity")
+                .AssertReply(ValidateTypingActivity, "check typing activity")
+                .AssertReply("Message sent after delay")
+                .StartTest();
+        }
+
+        [TestMethod]
+        [TestCategory("Middleware")]
+        public async Task ShowTyping_TestMiddleware_10SecondResponse_5SecondInterval()
+        {
+            TestAdapter adapter = new TestAdapter()
+                .Use(new ShowTypingMiddleware(500, 5000));
+
+            await new TestFlow(adapter, async (context) =>
+                {
+                    Thread.Sleep(10000);
```

Comment:
Similar comment.s

Reply:
Reduced timings accordingly to a couple of seconds max